### PR TITLE
Add segment events

### PR DESCRIPTION
**Pull request by Justin-Holmes**
https://www.pivotaltracker.com/story/show/154540864

### CODE REVIEW

**Comment by ellodanger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a linked tracker story for this PR in the description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: src/components/dialogs/BuyLinkDialog.js [hunk 1]
```diff
@@ -58,10 +61,12 @@ export default class BuyLinkDialog extends PureComponent {
     if (this.value.indexOf('http') !== 0) {
       this.value = `http://${this.value}`
     }
+    this.props.dispatch(trackEvent('buy-link-added-to-comment'))
```
Comment (alanpeabody) [inline src/components/dialogs/BuyLinkDialog.js]: Does this code not get fired for posts too? (component name seems generic)

Comment (Justin-Holmes) [reply src/components/dialogs/BuyLinkDialog.js]: Doh, good catch. It definitely does fire for posts.

**Comment by Justin-Holmes** [comment]
@alanpeabody Fixed 👌

File: src/sagas/analytics.js [hunk 2]
```diff
@@ -49,8 +49,15 @@ function* trackEvents() {
         break
       case ACTION_TYPES.COMMENT.CREATE_REQUEST: {
         const posts = yield select(selectPosts)
-        const artistInviteId = posts.get(get(action, 'payload.postId'), Immutable.Map()).get('artistInviteId')
-        yield put(trackEventAction('published_comment', artistInviteId ? { artistInviteId } : {}))
+        const postId = get(action, 'payload.postId')
+        const artGiveawayCategoryId = '48'
```
Comment (gblakeman) [inline src/sagas/analytics.js]: I’m guessing there’s a reason to hard code this `:id`? If so, just ignore me…

Comment (alanpeabody) [reply src/sagas/analytics.js]: @gblakeman this is so we can track participation is art-give aways. We don't really have a dynamic way to determine that unfortunately.

Comment (Justin-Holmes) [reply src/sagas/analytics.js]: @gblakeman Yeah, it's the id of the Art Giveaway category. I probably should've made it an ENV variable, but there shouldn't be a reason that the id should ever change.

**Pull request merged**